astropy/utils/data.py

```diff
@@ -175,7 +183,11 @@ def get_readable_fileobj(name_or_obj, encoding=None, cache=False,
         remote_timeout = conf.remote_timeout
 
     # Get a file object to the content
-    if isinstance(name_or_obj, six.string_types):
+    if isinstance(name_or_obj, PATH_TYPES):
+        # name_or_obj could be a Path object in Python 3
+        if six.PY3:
```

Comment:
Notice that this code becomes easier to understand:
```
        if HAS_PATHLIB:
            ...
```
This avoids one level of contextual knowledge that only Python 3 has pathlib.  So then the comment would be more like `# name_or_obj could be a Path object if pathlib is available`.